Module 4 Block 3 : OOP school library: add basic UI

aceDavon opened the pull request:
## In this PR, I have implemented the following tasks: :writing_hand: 

- [ ] Build interactive console apps.
- [ ] Run a program using the command line.
- [ ] Edit files permission to make the file executable

aceDavon committed:
Create Interactive classes

Create the main and the app files and link all required files

aceDavon committed:
Fix up methods codebase

aceDavon committed:
Make Main file executable

Modify permission on main.rb to make it executable

mahbubul14 commented on app.rb:
```diff
@@ -0,0 +1,103 @@
+    @rentals = []
+  end
+
+  def all_books
+    if @books.empty?
+      puts "There are no books created yet, Add some books to see them here :( \n\n"
+    else
+      @books.each { |bk| puts "\n Title: #{bk.title} by #{bk.author} \n \n" }
+    end
+  end
```
- [ ] In the school library console, when 1 is pressed it should be shown all the books from the list. But currently, when I press `"1"` it exits from the console. Kindly fix the issue. :+1: 

![list_all_books](https://user-images.githubusercontent.com/70483835/199808357-54f9749f-f8d5-4f59-9c77-6e8f8d569cb2.png)

aceDavon replied:
Noted and fixed. Thanks, for calling my attention to the issue

mahbubul14 reviewed (changes requested):
Hi @aceDavon,

Good job so far! :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

# Required Changes ♻️

- [ ] In your `README` file, Kindly update the getting started / setup section. This should be an example of how you may give instructions on setting up and running your project locally. The project is not running in the mentioned way.

- [ ] Kindly check the inline comments under the review. :arrow_down: 

###  To Highlights
:heavy_check_mark: No linter error.
:heavy_check_mark: Proper Github flow used.
:heavy_check_mark: Descriptive PR title & description.

### Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

aceDavon committed:
Fix string issue with a case on main.rb

Fix an issue with string type error on main.rb to stop exit on shell
unexpectedly

algerina reviewed (approved):
Hi @aceDavon , 
Good job so far! 

May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:


## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the changes requested by the previous TSEs
 
<img src="https://media0.giphy.com/media/lFaL6cWEGu2MGQeT7w/giphy.gif"/>

 ## Optional suggestions
 
 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

aceDavon committed:
Update startup documentation

Correct the Start up documentation on README.md to reflect how to properly
 start up the program after making the main.rb executable
